src/Listener.h

```diff
@@ -0,0 +1,39 @@
+/*
+ * File: Listener.h
+ * Author: Craig Cogdill
+ * Created: August 14, 2015 1:08PM
+ */
+
+#pragma once
+#include <vector>
+#include <string>
+#include <memory>
+
+class Alien;
+class Rifle;
+
+class Listener {
+   public:
+      Listener() = delete;
+      Listener(const std::string& notificationQueue, const std::string& handshakeQueue, const std::string& program);
+      virtual ~Listener() { Reset(); };
+      Listener(const Listener&) = delete;
+      Listener& operator=(const Listener&) = delete;
+      bool Initialize();
+      void Reset();
+      bool NotificationReceived();
+      bool SendConfirmation();
+      std::unique_ptr<Rifle> CreateFeedbackShooter();
```

Comment:
Why is this public. Only the public API should be public. Everything else should be private or protected.